### PR TITLE
Add org.gnome.Screenshot

### PR DESCRIPTION
**Pull request by A6GibKm**
Please review the cleanups. I saw that most of them were actually removed by flatpak builder so they might be non-redundant. Using https://gitlab.gnome.org/GNOME/gnome-screenshot/-/blob/master/org.gnome.Screenshot.json as a base.

### CODE REVIEW

**Comment by A6GibKm** [comment]
bot, build org.gnome.Screenshot

**Comment by A6GibKm** [comment]
Upstream issue: https://gitlab.gnome.org/GNOME/gnome-screenshot/-/issues/33

**Comment by Exalm** [comment]
I'm fine with maintaining it here, but I'm not sure it should be Flatpak-ed to begin with, since it won't work in e.g. KDE/Wayland, and we may in future remove the X11 backend or even move the whole UI to gnome-shell and make the app a dummy launcher for it: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2947

**Comment by A6GibKm** [comment]
The patches for the metainfo are already upstream. It might fail due to the size of the image.

**Comment by Exalm** [comment]
I don't think it makes sense to shoe 3.38 screenshot for 3.36: the UI is considerably different and the old screenshot matches better.

That said, I'll do 3.38 release, well, tomorrow.

File: org.gnome.Screenshot.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libhandy/0.91/libhandy-0.91.0.tar.xz",
+                    "sha256": "cfc1c1cc272d6932ef253fbed07b69a655b33e7c49fb66806beedb7199128c86"
+                }
+            ]
+        },
+        {
+            "name" : "gnome-screenshot",
+            "buildsystem" : "meson",
+            "builddir" : true,
```
Comment (Exalm) [inline org.gnome.Screenshot.json:73]: You shouldn't need this, it's only needed upstream for Builder.

File: org.gnome.Screenshot.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                "-Dexamples=false",
+                "-Dglade_catalog=disabled",
+                "-Dintrospection=disabled",
+                "-Dtests=false",
+                "-Dvapi=false"
+            ],
+            "sources" : [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libhandy/0.91/libhandy-0.91.0.tar.xz",
```
Comment (Exalm) [inline org.gnome.Screenshot.json:65]: There's 1.0.

Comment (Exalm) [reply org.gnome.Screenshot.json:65]: Speaking of which, I should totally bump the version.

**Comment by A6GibKm** [comment]
At the moment my priority is to be able to successfully build this, the 3.38-2 screenshot is mostly in line with the current app. Since 3.37 already has a correct metainfo the patch will be dropped anyways. 

issues:
- `appstream-util validate` fails on the new set of screenshot due to small size.
- 3.38 runtime (should) launches in a week. But to wait a week for the release is ok.

I am looking forward to the 3.38 release.

**Comment by A6GibKm** [comment]
bot, build org.gnome.Screenshot

**Comment by Exalm** [comment]
> appstream-util validate fails on the new set of screenshot due to small size. (tested on master's metainfo)

The old one was too small too. The app in general is too small, what can I do 🤷️

**Comment by A6GibKm** [comment]
The validation fails, but the flatpak builds just fine. The app can't be enlarged so it is fine, I just didn't knew if it would build 10 minutes ago.

**Comment by A6GibKm** [comment]
bot, build org.gnome.Screenshot

**Comment by A6GibKm** [comment]
bot, build org.gnome.Screenshot

**Comment by A6GibKm** [comment]
> but I'm not sure it should be Flatpak-ed to begin with

Silverblue users are currently use the flatpak in the fedora repos, this would allow user from other distros. I don't know if there is a better way to handle the cases when the flatpak won't work.

**Comment by Exalm** [comment]
TBH it should just be preinstalled on Silverblue, I don't know why it isn't. It's a core app.

**Comment by A6GibKm** [comment]
tl;dr newer versions install the flatpak version from the fedora repo during installation https://github.com/fedora-silverblue/issue-tracker/issues/34.

**Comment by A6GibKm** [comment]
whoops I missed the two PRs were this is discussed in flathub previously, #961 and #628.

**Comment by A6GibKm** [comment]
Needed to take screenshots in a sane way in Fedora Silverblue, there is a version on the `fedora` flatpak repository but there are many ways in which a flathub version would be superior.

I think the question of *why* is irrelevant as long as there is an interest, the more interesting question is whether it is worth it to have a flatpak given the drawback of it only being supported on GNOME+wayland, specially given that there is no clear way to communicate this to the end user.

I suspect that other desktop environments/distributions come with sane ways of taking screenshots out of the box and the only use-case for this is in GNOME.

**Comment by alatiera** [comment]
> I think the question of _why_ is irrelevant as long as there is an interest

Its not, Screenshot is not an app, but part of the Operating System. Its a core utility of GNOME. If your distro doesn't include it, or provide a replacement utility that's a bug with them and shouldn't really spill here. This looks like packaging something for the shake of packaging here.

**Comment by A6GibKm** [comment]
https://github.com/fedora-silverblue/issue-tracker/issues/34.

**Comment by Exalm** [comment]
To be clear, yes, Screenshot's upstream repo does include a flatpak manifest, but that's a development tool really, it exists for:

1. Builder, so that Run button works
2. CI

E.g. gnome-control-center also has a flatpak for similar purposes, and so do libraries like GTK.

**Comment by bilelmoussaoui** [comment]
I'm going to go ahead and close this one, see the other two merge requests for details. Silverblue should probably ship it as part of the base system and not as a Fltatpak neither.

**Comment by tpopela** [comment]
> Well ... if it is a core utility ... maybe Silverblue should have Gnome Screenshot installed as an RPM inside the vanilla image and not as a Flatpak.

From our point of view, we don't see Cheese as a core utility. It's another application and thus is available as a Flatpak in Silverblue. Shipping it as a Flatpak was approved by David King (the upstream Cheese maintainer). He actually created it as well - https://src.fedoraproject.org/flatpaks/cheese/c/18e64eb8e013f048227bc3d5bce5eb3614a810b1?branch=master

> (Because it is crazy to have an enormous runtime from `fedora` Flatpak repo just for Screenshot.)

It's not just for Screenshot. If you install the Silverblue (starting from Silverblue 32) you have plenty of Flatpaks preinstalled and sharing the same Fedora runtime. If user wants to avoid that he is free to overlay a Cheese RPM.

**Comment by tpopela** [comment]
I was pointed in the Silverblue ticket that I'm talking about Cheese, but the question is about Screenshot. Sorry about that. But it doesn't change anything about my answer.

**Comment by Exalm** [comment]
It does though, since Screenshot doesn't do anything on its own and talks to gnome-shell.

As mentioned, it may even become a dummy launcher for https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2947 in future.

**Comment by alatiera** [comment]
There is no issue with Silverblue providing screenshots as a flatpak ootb, but we can't really ship screenshots in Flathub.
